pkg/runs/runsGet_test.go

```diff
@@ -167,39 +168,66 @@ func NewRunsGetServletMock(t *testing.T, status int, runName string, runResultSt
 
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 
-		if r.URL.Path != "/ras/run" {
-			t.Errorf("Expected to request '/ras/run', got: %s", r.URL.Path)
-		}
-		if r.Header.Get("Accept") != "application/json" {
-			t.Errorf("Expected Accept: application/json header, got: %s", r.Header.Get("Accept"))
-		}
-		w.Header().Set("Content-Type", "application/json")
-		w.WriteHeader(status)
+		if strings.Contains(r.URL.Path, "/ras/runs/") {
+			if r.Header.Get("Accept") != "application/json" {
+				t.Errorf("Expected Accept: application/json header, got: %s", r.Header.Get("Accept"))
+			}
+			urlParts := strings.Split(r.URL.Path, "/")
+			runid := urlParts[3]
+			for _, runResult := range runResultStrings {
+				assert.Contains(t, runResult, runid)
+			}
 
-		values := r.URL.Query()
-		pageRequestedStr := values.Get("page")
-		runNameQueryParameter := values.Get("runname")
-		pageRequested, _ := strconv.Atoi(pageRequestedStr)
-		assert.Equal(t, pageRequested, 1)
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(status)
+			combinedRunResultStrings := ""
+			for index, runResult := range runResultStrings {
+				if index > 0 {
+					combinedRunResultStrings += ","
+				}
+				combinedRunResultStrings += runResult
+			}
 
-		assert.Equal(t, runNameQueryParameter, runName)
+			w.Write([]byte(fmt.Sprintf(`
+				%s 
+			`, combinedRunResultStrings)))
 
-		combinedRunResultStrings := ""
-		for index, runResult := range runResultStrings {
-			if index > 0 {
-				combinedRunResultStrings += ","
+		} else {
+			if r.URL.Path != "/ras/runs" {
+				t.Errorf("Expected to request '/ras/runs', got: %s", r.URL.Path)
+			}
+			if r.Header.Get("Accept") != "application/json" {
+				t.Errorf("Expected Accept: application/json header, got: %s", r.Header.Get("Accept"))
+			}
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(status)
+
+			values := r.URL.Query()
+			pageRequestedStr := values.Get("page")
+			runNameQueryParameter := values.Get("runname")
+			pageRequested, _ := strconv.Atoi(pageRequestedStr)
+			assert.Equal(t, pageRequested, 1)
+
+			assert.Equal(t, runNameQueryParameter, runName)
+
+			combinedRunResultStrings := ""
+			for index, runResult := range runResultStrings {
+				if index > 0 {
+					combinedRunResultStrings += ","
+				}
+				combinedRunResultStrings += runResult
 			}
-			combinedRunResultStrings += runResult
+
+			w.Write([]byte(fmt.Sprintf(`
+			 {
+				 "pageNumber": 1,
+				 "pageSize": 1,
+				 "numPages": 1,
+				 "amountOfRuns": %d,
+				 "runs":[ %s ]
+			 }`, len(runResultStrings), combinedRunResultStrings)))
 		}
```

Comment:
method getting too big. Seems that it could be split into one method which decides what to do, and 2 new methods which actually do things ?